### ISSUE TITLE
Step over not working in v0.9.9

### ISSUE THREAD

**Issue by MohitSethi99** [EpicGamesExt/raddebugger#190]
Step over is working totally fine in v0.9.8

https://github.com/EpicGamesExt/raddebugger/assets/29519295/e2439575-44e1-4ff0-ad07-a663bdf2e6ff

While in v0.9.9, the step over doesn't seem to be working, sometimes the target is stuck and status in raddbg says running.

https://github.com/EpicGamesExt/raddebugger/assets/29519295/7556dd5b-b071-4e67-8ccc-c4fdd991f916

**Comment by MohitSethi99** [comment]
Tried deleting all the RADDBGI files, no luck.

**Comment by MohitSethi99** [comment]
It's not the for loop.
It's doing this in my hobby project and not able to repro it on work project.

Here is the project in case: https://github.com/MohitSethi99/ArcGameEngine
Branch: dx12
Run: scripts/GenerateSolution.bat
PS: It will require dotnet 7 components from VS installer

**Comment by MohitSethi99** [comment]
Yes, always failing at the same location.

**Comment by ryanfleury** [comment]
Since this is a hobby project, can you just send me a build? I am running into issues almost ASAP getting it set up, and I think it would just be faster to have a prebuilt binary I can test with locally.

**Comment by MohitSethi99** [comment]
Definitely, will send it as soon as I reach home. (1~2hrs)

**Comment by MohitSethi99** [comment]
Here you go: https://drive.google.com/file/d/1lCLeUtlpSvuQvlkNH1gyEkV4jfPE2d4j/view?usp=sharing

Exe: Arc-Editor.exe
Upon launch it will ask for project to open, simply clicking cancel should open the editor with no project loaded.
A good place to check out is `while` loop in `ArcEngine::Application::Run()`

**Comment by ryanfleury** [comment]
Thanks! I'll take a look.

**Comment by MohitSethi99** [comment]
Turns out remedybg is behaving the same for "step into" and not taking to the correct location. I'll investigate further and will update here.

**Comment by MohitSethi99** [comment]
The issue is happening when I'm calling a function in a dll. specifically nethost.dll I use to enable scripting backend in my engine. It allows to interop with C#. More details here: https://learn.microsoft.com/en-us/dotnet/core/tutorials/netcore-hosting
Will create a small example by the weekend.

**Comment by MohitSethi99** [comment]
Here is the visual studio solution for the minimal reproduction: https://drive.google.com/file/d/1rFZjYVJ2vn9ABg3JkbCZ7uOAJgL4RElt/view?usp=sharing

Zip is bloated cos of dotnet files. output directory has "App.runtimeconfig.json" config file and ".dotnet" folder which contains the dotnet framework that will be loaded by the code in Main.cpp

![image](https://github.com/EpicGamesExt/raddebugger/assets/29519295/4d8bcc0a-e737-472f-a899-9e440c9dcc11)
According to Visual Studio output, an exception is thrown (presumably in dotnet library) when Line 39 is executed. Rest of the code execution is fine in VS and remedy. But raddbg's step_over and step_into breaks after Line 40.

**Comment by MohitSethi99** [comment]
Issue is observed after this particular exception: 
```Exception thrown at 0x00007FFE0252543C (KernelBase.dll) in Arc-Editor.exe: 0x04242420 (parameters: 0x0000000031415927, 0x00007FFD20B90000, 0x0000000C5270E670).```

![image](https://github.com/EpicGamesExt/raddebugger/assets/29519295/a38a6c50-9346-408d-8c51-ef36aed27fb5)

**Comment by MohitSethi99** [comment]
Also noticed that this is written to the RadDebugger's output view:
```
onecore\vm\dv\storage\plan9\rdr\dll\util.cpp(99)\p9np.dll!00007FF9B890F0CC: (caller: 00007FF9B89093B0) LogHr(1) tid(1bf8) C0000034     Msg:[?????????????????????????????????????????????????????/??????????????????????????????????????????????)] 
```

**Comment by ryanfleury** [comment]
Sorry, I've been a bit swamped with other stuff. Can you somehow resend that minimal repro to me? Google removed it due to "terms of service violations"... sigh...

**Comment by MohitSethi99** [comment]
I hope onedrive doesn't remove it.
https://1drv.ms/u/s!ApjsIRdEL5T4kxab7q9YlvPVLww_?e=n4jIby

**Comment by MohitSethi99** [comment]
I think #106 is related, not sure though.

**Comment by ryanfleury** [comment]
I think *probably* not, but not clear... Just fixed that one again in 489ae56223210c3ceefe39a899c34f8b4c9ffb54, digging into this one now.

**Comment by ryanfleury** [comment]
Well, that took way longer than I expected, and it took way longer than it deserved! This was being caused by an incredibly stupid state machine bug I introduced during a pass over the Demon layer between 0.9.8 and 0.9.9, and it was difficult to locate. This should be fixed in 435877824a7fcc56ad1d860c700d85685a7e944a.

**Comment by MohitSethi99** [comment]
Thanks a lot! It works now :)
